src/Modal.js

```diff
@@ -41,7 +41,7 @@ module.exports = {
         var closeDialog = function() {
             ReactDOM.unmountComponentAtNode(self.getOrCreateContainer());
 
-            if (props && props.onFinished) props.onFinished.apply(null, arguments);
+            if (props && props.onFinished) props.onFinished.call(null, false);
```

Comment:
Looks like that code was added in d938ba70d38c81cd2fcfa2be4b0fb852814dd7d8, which also carries this bug (and does not use `arguments` in any way)